lib/Mojo/WebSocketProxy/Backend/JobAsync.pm

```diff
@@ -111,30 +111,30 @@ sub call_rpc {
     $self->{client}->start->get;
 
     $req_storage->{call_params} ||= {};
-
     my $rpc_response_cb = $self->get_rpc_response_cb($c, $req_storage);
 
     my $before_get_rpc_response_hook = delete($req_storage->{before_get_rpc_response}) || [];
     my $after_got_rpc_response_hook  = delete($req_storage->{after_got_rpc_response})  || [];
     my $before_call_hook             = delete($req_storage->{before_call})             || [];
     my $params = $self->make_call_params($c, $req_storage);
     $log->debugf("method %s has params = %s", $method, $params);
-
     $_->($c, $req_storage) for @$before_call_hook;
-
     $self->client->submit(
         name   => $req_storage->{name},
-        params => encode_json_utf8($params)
+        params => encode_json_utf8($params),
+        rpc_queue_client_tv => [Time::HiRes::gettimeofday],
```

Comment:
what does `tv` stands for? Can we use descriptive name, please?

Reply:
i dont think we should be adding this extra parameter, to get stats that we actually already have.
we already have hooks in Websocket that will record for us the time spent in each call. In fact you can easily see now how long reset_password request takes to be served (attached screenshot)

<img width="1572" alt="Screenshot 2019-07-16 at 2 24 00 PM" src="https://user-images.githubusercontent.com/24886012/61271019-0168dc00-a7d6-11e9-96fc-9d20f1ca8b4d.png">

so I think in here we should only have one `stats_inc` indicating a successful request to the queue.

Reply:
@nael-binary I'm trying to cover what is missing in the existing metrics: `queue latency` (i.e. the interval between the moment a message is queued, until it is dequeued. I'm not sure if it is the correct term). How can we calculate it from the existing metrics? The closest (not exactly correct) answer is: `bom_websocket_api.v_3.rpc_call.timing` minus `bom_websocket_api.v_3.rpc_call.timing`.

Reply:
in this current way what its really measuring the network latency only as in (how long it took us from the moment we sent the request to the queue, until the worker picked it up. -not including the processing time.
I believe those are the two things that we are interested in, in terms of timing only:
- time spent in processing each job in worker (has to be added)
- time spent in order to serve the request to client. Hooks in WS are already in place for that.

why `bom_websocket_api.v_3.rpc_call.timing` is not correct? isnt it giving us the time spent on each RPC call, whether from normal RPC or job_async one?

Reply:
@nael-binary I edited the last sentence of my message: 
```
The closest (not exactly correct) answer is: bom_websocket_api.v_3.rpc_call.timing minus bom_rpc.v_3.call.timing.
```
Anyway,I admit that keeping with the old metrics is more intuitive if we want to make before/after comparisons. We can measure queue performance by the existing metrics fairly (we cannot separate request queue latency from response latency, for example). 

It means that I can close two RPs. Hurray!